<launch>

    <node pkg="location" type="locationV2" name="locationV2"  output="screen">
        <param name="mapPCDFileName" value="$(find location)/map/fps6/yunshenchu.pcd"/>
        <param name="lidarName" value="velodyne_points"/>
        <param name="enableSwitchLidarData" value="false"/>
        <param name="switchBufferSize" value="4"/>
        <param name="enablePublishGlobalDenseMap" value="false"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0.0 0 0.59 0 0 0 base_link velodyne 50" />
    <node pkg="tf" type="static_transform_publisher" name="link3_broadcaster" args="0.0 0 0.54 0 0 0 base_link imu 50" />
    <node pkg="tf" type="static_transform_publisher" name="link4_broadcaster" args="0.0 0 0.0 0 0 0 map odom 50" />
<!--    <node pkg="tf" type="static_transform_publisher" name="link5_broadcaster" args="0.0 0.0 -0.13 -1.568 -0.0 -1.88 velodyne camera_link 50" />-->

    <node name="MapServer" pkg="map_server" type="map_server" output="screen" args="$(find location)/map/fps6/yunshenchu.yaml"/>

    <node pkg="detection" type="detection" name="detection" output="screen">
        <param name="cameraFrame" value="AreoCameraFront_depth_optical_frame"/>
        <param name="cameraPoint" value="/AreoCameraFront/depth/color/points"/>
        <param name="odomFrame" value="odom"/>
        <param name="FilterFieldName" value="z"/>
        <param name="setFilterLimitsUp" value="5.0"/>
        <param name="setFilterLimitsDown" value="0.0"/>
        <param name="use_static_tf" value="true"/>
    </node>

    <node pkg="detection" type="rawmapping" name="rawmapping" output="screen">
        <param name="cameraFrame" value="AreoCameraFront_depth_optical_frame"/>
        <param name="mpaFrame" value="odom"/>
        <param name="cameraPoint" value="/AreoCameraFront/depth/color/points"/>
        <param name="FilterFieldName" value="z"/>
        <param name="setFilterLimitsUp" value="3.5"/>
        <param name="setFilterLimitsDown" value="0.3"/>
        <param name="setFilterheight" value="0.1"/>
        <param name="setOutlierFilter" value="false"/>
        <param name="setOutlierFilterMeanK" value="10"/>
        <param name="setOutlierFilterThresh" value="1.0"/>
        <param name="voxelsize" value="0.04"/>
        <param name="roshz" value="1000"/>
        <param name="wait_surfel_finished" value="10000"/>
        <param name="setPointNumLimit" value="500000"/>
    </node>

    <!-- Get Pointcloud and transform to Gridmap-->
<!--    <node pkg="detection" type="grid_map_test" name="grid_map_test" output="screen" >-->
<!--        <param name="point_sub_topic" value="/fusemap" />-->
<!--        <param name="grid_pub_topic" value="/grid_map" />-->
<!--        <param name="grid_pub_topic" value="/odom" />-->
<!--        <param name="grid_map_size_x" value="20.0" />-->
<!--        <param name="grid_map_size_y" value="20.0" />-->
<!--        <param name="grid_map_resolution" value="0.5" />-->
<!--    </node>-->

    <arg name="rviz" default="true" />
    <param name="use_sim_time" value="true" />
    <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find location)/rviz_cfg/ndt_matching.rviz" />
    </group>


</launch>
